### DISCUSSION TITLE
Upgrade problems; Online migration expected to match one row error

### DISCUSSION THREAD

**Discussion by kingkong-cmd** [miguelgrinberg/Flask-Migrate#485] [Q&A]
I tried to do a new migration on my local postgres dev DB, and got the "_Target database is not up to date_" error. So I tried to run **flask db upgrade** which gave me the following error:

`ERROR [flask_migrate] Error: Online migration expected to match one row when updating '3fba6dfbb433' to '0db152ce17b6' in 'alembic_version'; 0 found`

(_0db152ce17b6_ is the previous migration before I tried to create a new one today)

So I tried **flask db heads**, **flask db history** and **flask db current** which gave me the results below. I'm not sure how to interpret  the above message together with the below output. In addition, when I run my app, everything seems to work as if the upgrade has been done. On my prod DB the upgrade seems to have been done successfully. Can anyone help me figure out what's going on?

```
flask db heads
0db152ce17b6 (head)
```

```
flask db current: 
3fba6dfbb433
```

```
flask db history
3fba6dfbb433 -> 0db152ce17b6 (head), empty message
b1cb58b9ca83 -> 3fba6dfbb433, empty message
4b27b6043409 -> b1cb58b9ca83, empty message
88f37cb5274b -> 4b27b6043409, empty message
```

**Comment by miguelgrinberg** [comment]
This is an alembic error, so not directly to this package. If you want the best possible advice then you should ask in an Alembic forum. But in any case, the error suggests your `alembic_version` table is bad.

Before you continue trying things out I would suggest you make a back up, just in case.

Then you can try `flask db stamp 0db152ce17b6`. If you are 100% sure the contents of this migration do exist in your database, then maybe the stamp command will fix your version table and then you will be able to create another migration on top.

**Reply by kingkong-cmd** [reply]
Ok, thanks and understood. The contents of the migration does not exist in the db, so I'll try my luck in the alembic github.

**Reply by miguelgrinberg** [reply]
Then you can do `flask db stamp 3fba6dfbb433` and then attempt the upgrade again. Of course do not try anything before making a backup!
